### ISSUE TITLE
`unnecessary_sort_by` FN on field access

### ISSUE THREAD

**Issue by profetia** [rust-lang/rust-clippy#16405]
### Summary

`unnecessary_sort_by` FN on field access

### Lint Name

unnecessary_sort_by

### Reproducer

I tried this code:

```rust
fn foo() {
    let mut v: Vec<(i32, &str)> = vec![(1, "foo"), (2, "bar")];
    // v.sort_by(|(_, s1), (_, s2)| s1.cmp(s2));

    v.sort_by(|a, b| a.0.cmp(&b.0));

    struct Item {
        key: i32,
        value: String,
    }

    let mut items = vec![
        Item {
            key: 2,
            value: "b".to_string(),
        },
        Item {
            key: 1,
            value: "a".to_string(),
        },
    ];
    items.sort_by(|item1, item2| item1.key.cmp(&item2.key));
}
```

I expected to see this happen:
```
warning: consider using `sort_by_key`
 --> src/main.rs:9:5
  |
9 |     v.sort_by(|a, b| a.0.cmp(&b.0));
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `v.sort_by_key(|a| a.0)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_sort_by
note: the lint level is defined here
 --> src/main.rs:3:9
  |
3 | #![warn(clippy::unnecessary_sort_by)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: consider using `sort_by_key`
  --> src/main.rs:26:5
   |
26 |     items.sort_by(|item1, item2| item1.key.cmp(&item2.key));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `items.sort_by_key(|item1| item1.key)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_sort_by
```

Instead, this happened:
Nothing

### Version

```text
rustc 1.94.0-nightly (fecb335cb 2026-01-07)
binary: rustc
commit-hash: fecb335cbad3d84ef3da39191ba094e6c726a5b4
commit-date: 2026-01-07
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.8
```
